db/db_impl/db_impl_open.cc

```diff
@@ -622,6 +622,15 @@ Status DBImpl::Recover(
           cfd->CreateNewMemtable(*cfd->GetLatestMutableCFOptions(),
                                  kMaxSequenceNumber);
         }
+      } else if (immutable_db_options_.track_and_verify_wals_in_manifest) {
```

Comment:
If the recovered memtables are flushed, then the WAL obsoletion event will be logged in MANIFEST when logging the flush.

But in https://github.com/facebook/rocksdb/blob/master/db/db_impl/db_impl_open.cc#L1192, if the recovered memtable is empty, the CF's log number will also be updated, which may cause some WALs to become obsolete due to the same reason as updating log number of empty WAL in db_impl_write.cc.

Reply:
We could consider moving this chunk to `RecoverLogFiles` (which already logs `VersionEdit`s).